### PR TITLE
Merge EXTRA_EXPORTED_RUNTIME_METHODS

### LINKED ISSUE #4375: EXTRA_EXPORTED_RUNTIME_METHODS not considered for functions exported from library_fs.js
Hi,

When trying to add FS exported functions (such as `FS_writeFile` which is not exported by default), adding them to `EXTRA_EXPORTED_RUNTIME_METHODS` has no effect since only `EXPORTED_RUNTIME_METHODS` is considered there.

See line 8 of library_fs.js :
https://github.com/kripken/emscripten/blob/incoming/src/library_fs.js#L8

Maybe it would be better to merge the two list at the beginning of the code so that everything is considered everywhere ?

### PR DESCRIPTION
**Pull request by AlexPerrot**
This PR adds merging of `EXTRA_EXPORTED_RUNTIME_METHODS` with `EXPORTED_RUNTIME_METHODS` to ensure function in `EXTRA` are exported by libraries looking only at `EXPORTED_RUNTIME_METHODS`, like `FS`.

Fixes #4375

### CODE REVIEW

**Comment by kripken** [comment]
Looks good, thanks, but let's add a test for this. I don't remember offhand what use case this fixes so I don't have a suggestion for where to add it, but let me know if you need help thinking about that.

**Comment by AlexPerrot** [comment]
The use case was that specifying `-s EXTRA_EXPORTED_RUNTIME_FUNCTIONS=["FS_writeFile"]` didn't do anything since `library_FS` only considers `EXPORTED_RUNTIME_METHODS`.

I manually tested it by compiling a program with `-s EXTRA_EXPORTED_RUNTIME_FUNCTIONS=["FS_writeFile"]` and verifying that `Module.FS_writeFile` exists. In which test suite should I add such a test ?

Also, I wonder if this could cause problems for libraries looking at `EXTRA_EXPORTED_RUNTIME_FUNCTIONS`. Maybe we should also empty it.

**Comment by kripken** [comment]
Somewhere in `tests/test_other.py` probably is best. A random example test that does something similar (checks an `-s` option) is `test_node_catch_exit`, so something similar in general form maybe.

I guess emptying it makes sense, although it's just meant for internal usage, so I don't think we need to - no library should look at it.

**Comment by AlexPerrot** [comment]
I added a test based on `test_node_catch_exit`.

I looked at libraries, and indeed, none is using `EXTRA`.

File: tests/test_other.py [hunk 1]
```diff
@@ -2637,6 +2637,37 @@ def test_node_catch_exit(self):
     self.assertContained(reference_error_text,
                          run_js ('index.js', engine=NODE_JS, stderr=STDOUT, assert_returncode=None))
 
+  def test_extra_exported_methods(self):
+    # Test with node.js that the EXTRA_EXPORTED_RUNTIME_METHODS setting is considered by libraries
+    if NODE_JS not in JS_ENGINES:
+      return
```
Comment (kripken) [inline tests/test_other.py]: can write `return self.skip("..reason..")` (moving the explaining comment into the reason; then it's shown when skipped)

lgtm aside from that, thanks.

**Comment by kripken** [comment]
Looks like the last comment here was a minor fix before merging. @AlexPerrot?

**Comment by AlexPerrot** [comment]
Sorry, I forgot about this.
This should be good now.

**Comment by kripken** [comment]
Thanks!

**Pull request merged**